src/sentry/conf/types/kafka_definition.py

```diff
@@ -142,6 +142,15 @@ class ConsumerDefinition(TypedDict, total=False):
 
     stale_topic: Topic
 
+    # How long to wait for the consumer to shut down.
+    #
+    # By default, the join timeout of a consumer is 25 seconds. None means no
+    # timeout.
+    #
+    # Override this if the consumer is not idempotent and takes longer to shut
+    # down, at the risk of stability issues.
+    default_join_timeout: float | None
```

Comment:
We cannot guarantee exactly once by increasing the join timeout if the consumer takes more than 30 seconds to flush. In that case Kafka will not let us commit, thus going back to at least once. 
I would not state that, increasing the join timeout, allows you to preserve exactly once if the consumer takes too long to shut down.